ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BaseClusterRequest.java

```diff
@@ -53,6 +55,11 @@
 
   protected ProvisionAction provisionAction;
 
+  /**
+   * The raw request body. We would like to persist it.
+   */
+  protected String rawRequestBody;
```

Comment:
Persisting the cluster creation template is a great idea, but we probably need to filter some information out of the template first (the "default_password" field, any configuration items in the cluster config that are of type "PASSWORD") prior to persisting the document.  

There may be other types of information that should be filtered out as well, but from a security perspective, we should definitely filter out any password information prior to persisting the data.